### PR TITLE
convert nested tree to flat tree

### PR DESCRIPTION
**Pull request by ydomenjoud**

### CODE REVIEW

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-tree/components/kal-tree-node.component.ts [hunk 1]
```diff
@@ -29,55 +28,49 @@ import { KalTreeNode } from '../classes/kal-tree-node';
  */
 @Component({
   selector: 'kal-tree-node',
-  template: '<ng-content></ng-content>',
+  template: '<ng-content ></ng-content>',
```
Comment (jsp-up) [inline projects/kalidea/kaligraphi/src/lib/atoms/kal-tree/components/kal-tree-node.component.ts]: remove space

File: src/app/atoms/tree/tree.component.html [hunk 2]
```diff
@@ -2,22 +2,42 @@ <h1>Treeview</h1>
 
 <div class="example">
 
-  <kal-tree [dataSource]="data" [treeControl]="treeControl">
+  <kal-tree [dataSource]="dataSource" [treeControl]="treeControl" (selectionChanged)="changed($event);">
 
-    <kal-tree-node *kalTreeNodeDef="let node">
-      {{node.title}}
+    <kal-tree-node *kalTreeNodeDef="let node" [kalDrag]="node" (kalDrop)="drop($event)" kalTreeNodePadding>
+      {{node.title}} qsqsd
```
Comment (jsp-up) [inline src/app/atoms/tree/tree.component.html]: qsqsd

File: src/app/atoms/tree/tree.component.ts [hunk 3]
```diff
@@ -45,13 +57,36 @@ const TREE_DATA: KalTreeNode[] = [
...
+
+  changed($event) {
+    // console.log($event);
+  }
 
-  hasNestedChild = (_: number, nodeData: KalTreeNode) => {
-    return !!nodeData.children;
-  };
+  drop($event) {
+    console.log($event);
```
Comment (jsp-up) [inline src/app/atoms/tree/tree.component.ts:82]: console.log

**Review by jsp-up — CHANGES_REQUESTED** [review]

File: src/app/atoms/tree/tree.component.ts (hunk 3 above)
Comment (ydomenjoud) [reply src/app/atoms/tree/tree.component.ts:82]: Not an issue in playground

**Review by jsp-up — APPROVED** [review]

**Pull request merged**